pkg/kube/client/client.go

```diff
@@ -120,6 +143,21 @@ func configForProvider(ctx context.Context, local client.Client, providerConfigN
 					return nil, errors.Wrap(err, errInjectAzureCredentials)
 				}
 			}
+		case kconfig.IdentityTypeUpboundToken:
+			switch id.Source { //nolint:exhaustive
+			case xpv1.CredentialsSourceInjectedIdentity:
+				return nil, errors.Errorf("%s is not supported as identity source for identity type %s",
+					xpv1.CredentialsSourceInjectedIdentity, kconfig.IdentityTypeUpboundToken)
+			default:
+				tkn, err := resource.CommonCredentialExtractor(ctx, id.Source, b.local, id.CommonCredentialSelectors)
```

Comment:
nit: static or exchangeable token

Reply:
Could you clarify, couldn't get.

Reply:
sorry: tried to suggest that the variable being returned `tkn` could be called `staticToken` or `exchangeableToken` to it's clear this is the "before" state.